rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,429 @@
+
+- Implementation details are not specified. The RFC shepherd group has some (feature incomplete) POC's sufficient for a generic specification. (See [Native support in Nix](#Native-support-in-Nix) )
+
+## Current State
+
+A third-party tool called [nixdoc](https://github.com/nix-community/nixdoc) has emerged, which codifies its own rules as to the internal and external formats of a Nix doc-comment. This tool has seen some adoption, notably for the `nixpkgs.lib` functions.
+
+Here is an example of the format understood by *nixdoc*:
+
+```nix
+  # nixpkgs/lib/trivial.nix
+
+  /* The constant function
+
+     Ignores the second argument. If called with only one argument,
+     constructs a function that always returns a static value.
+
+     Type: const :: a -> b -> a
+     Example:
+       let f = const 5; in f 10
+       => 5
+  */
+  const =
+    # Value to return
+    x:
+    # Value to ignore
+    y: x;
+```
+
+## Current problems
+
+### Multiplicity of formats
+
+Within nixpkgs alone, several conventions for doc-comments have emerged, see [1], [2] and [3].
+
+Notably, most doc-comments utilize some fraction of the of CommonMark syntax, even if they are not meant to be rendered.
+
+[1]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/trivial-builders/default.nix
+[2]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/make-derivation.nix
+[3]: https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/make-disk-image.nix
+
+In general, the format for writing documentation strings is **not formally specified**.
+
+Among the formats encountered in the wild, the one used in `nixpkgs/lib` is the only one intended to be rendered as part of an API documentation, via the nixdoc third-party tool, whose syntax has not been standardized.
+
+### Impossible to differentiate from internal comments
+
+The lack of a formal definition of a doc-comment also means there is no reliable way to distinguish them from internal comments, which would result in automatically-produced API documentation which includes the wrong type of comments.
+
+### References to the problems above
+
+> This curated link collection highlights the Nix ecosystem's inconsistencies, a primary focus of this RFC.
+
+#### nixpkgs - comment examples
+
+- [lib/attrsets](https://github.com/NixOS/nixpkgs/blob/master/lib/attrsets.nix)
+- [trivial-builders](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/trivial-builders.nix)
+- [stdenv/mkDerivation](https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/make-derivation.nix)
+- [nixos/lib/make-disk-image](https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/make-disk-image.nix)
+
+# Design
+[design]: #detailed-design
+
+In the following we give a comprehensive overview to our decision that we've made. Detailed arguments for and against every decision can be found in the [Decisions](#Decisions) section
+
+## CommonMark
+
+**CommonMark is the content of all doc-comments.**
```

Comment:
```suggestion
**CommonMark is the format of the content of all doc-comments.**
```

Reply:
Does this mean keywords like `@param` are excluded from the spec?

Reply:
Yep. For now we leave it just tooling specific implementation. We can make another rfc in the future this one is quite complicated already.

Reply:
Or just 
```suggestion
**Doc comments are formatted in [CommonMark](https://commonmark.org/).**
```